scripts/sanity_chk/sanitylib.py

```diff
@@ -2315,8 +2315,8 @@ def extract_overlays(args):
 
         overlays = extract_overlays(args)
 
-        if (self.testcase.extra_configs or self.coverage or
-                self.asan or self.ubsan):
+        if os.path.exists(os.path.join(instance.build_dir,
+                                       "sanitycheck", "testcase_extra.conf")):
```

Comment:
WIll the use case stay the same? I am not familiar with this usage of sanitycheck, hence my questions. Is the existence of testcase_extra.conf enough to get it counted? Is it possible that the testcase_extra.conf already exists (e.g. from previous runs) but adding it is actually not correct?

Reply:
If the issue is that the file not always exist, why not modifying the if logic by adding `and os.path.exists(...)`

Reply:
`sanitycheck` will always build from scratch, so there is no reusing.
If you notice the linked commit: https://github.com/zephyrproject-rtos/zephyr/commit/51d6931b1600907b4d4ba0608cf2295f4dd5cdd0

then that commit changed the existing behavior from always producing the file, to only produce the file under certain circumstances, therefore this PR follows up on that change, to check the existence of the file, as it is not generated in all situations (and thus it should not be applied if not generated)